### PR TITLE
Iterators, take two

### LINKED ISSUE #77: Fix aborting on end
We get random errors (screenshot from Sauce Labs):

image

### PR DESCRIPTION
**Pull request by vweevers**
Closes #77 (and the 4 problems described there) and properly handles errors.

### CODE REVIEW

File: iterator.js [hunk 1]
```diff
@@ -52,32 +53,40 @@ util.inherits(Iterator, AbstractIterator)
...
-    }
+
+    // If an error occurs, the transaction will abort and we can
+    // get the error from "transaction.error".
+    onError: noop
   })
+
+  // Override IDBWrapper's event handlers for a simpler flow.
+  self.transaction.oncomplete = self.transaction.onabort = function () {
+    self.onComplete()
```
Comment (ralphtheninja) [inline iterator.js]: :+1:

Comment (ralphtheninja) [reply iterator.js]: Or `= self.onComplete.bind(self)` but nit pick

File: test/manual/end-test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+})
+
+test('repeated end', function (t) {
+  function loop (n) {
+    if (n > 5e3) return t.end()
+
+    var iterator = db.iterator()
+
+    iterator.next(function (err, key, value) {
+      if (err) throw err
```
Comment (ralphtheninja) [inline test/manual/end-test.js]: I presume these throw statements is to not spam the tests too much?

File: iterator.js (hunk 1 above)
Comment (vweevers) [reply iterator.js]: I figured it's more convenient to override if we want to unit test it

File: test/manual/end-test.js (hunk 2 above)
Comment (vweevers) [reply test/manual/end-test.js]: Yeah, in an earlier iteration I got thousands of errors ;)

**Pull request merged**